Sources/NetworkProtection/Diagnostics/NetworkProtectionTunnelFailureMonitor.swift

```diff
@@ -39,137 +39,64 @@ final public class NetworkProtectionTunnelFailureMonitor {
 
     private static let monitoringInterval: TimeInterval = .seconds(10)
 
-    public var publisher: AnyPublisher<Result, Never> {
-        failureSubject.eraseToAnyPublisher()
-    }
-    private let failureSubject = PassthroughSubject<Result, Never>()
-
-    private actor TimerRunCoordinator {
-        private(set) var isRunning = false
-
-        func start() {
-            isRunning = true
-        }
-
-        func stop() {
-            isRunning = false
+    @MainActor
+    private var task: Task<Never, Error>? {
+        willSet {
+            task?.cancel()
         }
     }
 
-    private var timer: DispatchSourceTimer?
-    private let timerRunCoordinator = TimerRunCoordinator()
-    private let timerQueue: DispatchQueue
+    @MainActor
+    var isStarted: Bool {
+        task?.isCancelled == false
+    }
 
     private let tunnelProvider: PacketTunnelProvider
```

Comment:
Should this be `weak` to prevent a retain cycle?

Reply:
This one's important as it may also explain why the timer wasn't being released before.